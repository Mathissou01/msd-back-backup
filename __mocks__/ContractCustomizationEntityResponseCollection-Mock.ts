export const contractCustomizationEntityResponseCollectionMock = {
  data: [
    {
      attributes: {
        homepage: {
          data: {
            attributes: {
              quizAndTipsBlock: {
                data: {
                  id: "1",
                  attributes: {
                    title: "Quiz & Astuces",
                    displayBlock: false,
                    displayQuiz: false,
                    quiz: {
                      data: {
                        id: "2",
                        attributes: {
                          title: "Nom du quiz",
                          publishedAt: "2022-11-20T02:34:14.535Z",
                        },
                      },
                    },
                    displayTips: false,
                    tips: {
                      data: [],
                    },
                  },
                },
              },
            },
          },
        },
      },
    },
  ],
};
