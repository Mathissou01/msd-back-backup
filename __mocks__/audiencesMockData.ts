import { GetAudiencesByContractIdDocument } from "../src/graphql/codegen/generated-types";

export const getNoAudienceMockedData = [
  {
    request: {
      query: GetAudiencesByContractIdDocument,
      variables: {
        filters: {
          contract: {
            id: {
              eq: undefined,
            },
          },
          isActive: {
            eq: true,
          },
        },
      },
    },
    result: {
      data: {
        audiences: {
          data: [],
        },
      },
    },
  },
];

export const getSingleAudienceMockedData = [
  {
    request: {
      query: GetAudiencesByContractIdDocument,
      variables: {
        filters: {
          contract: {
            id: {
              eq: undefined,
            },
          },
          isActive: {
            eq: true,
          },
        },
      },
    },
    result: {
      data: {
        audiences: {
          data: [
            {
              id: 1,
              attributes: {
                isActive: true,
                type: "Particuliers",
                __typename: "Audience",
              },
            },
          ],
        },
      },
    },
  },
];

export const getTwoAudiencesMockedData = [
  {
    request: {
      query: GetAudiencesByContractIdDocument,
      variables: {
        filters: {
          contract: {
            id: {
              eq: undefined,
            },
          },
          isActive: {
            eq: true,
          },
        },
      },
    },
    result: {
      data: {
        audiences: {
          data: [
            {
              id: 1,
              attributes: {
                isActive: true,
                type: "Particuliers",
                __typename: "Audience",
              },
            },
            {
              id: 2,
              attributes: {
                isActive: true,
                type: "Collectifs",
                __typename: "Audience",
              },
            },
          ],
        },
      },
    },
  },
];

export const getAllAudiencesMockedData = [
  {
    request: {
      query: GetAudiencesByContractIdDocument,
      variables: {
        filters: {
          contract: {
            id: {
              eq: undefined,
            },
          },
          isActive: {
            eq: true,
          },
        },
      },
    },
    result: {
      data: {
        audiences: {
          data: [
            {
              id: 1,
              attributes: {
                isActive: true,
                type: "Particuliers",
                __typename: "Audience",
              },
            },
            {
              id: 2,
              attributes: {
                isActive: true,
                type: "Collectifs",
                __typename: "Audience",
              },
            },
            {
              id: 3,
              attributes: {
                isActive: true,
                type: "Professionnels",
                __typename: "Audience",
              },
            },
          ],
        },
      },
    },
  },
];
