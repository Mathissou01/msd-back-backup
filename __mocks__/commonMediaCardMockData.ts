import { GetFilesPaginationByFolderIdDocument } from "../src/graphql/codegen/generated-types";

export const defaultMockData = [
  {
    request: {
      query: GetFilesPaginationByFolderIdDocument,
      filters: {
        folder: {
          id: {
            eq: 2,
          },
        },
      },
      pagination: {
        page: 1,
        pageSize: 10,
      },
      sort: "createdAt:desc",
    },
    result: {
      data: {
        uploadFiles: {
          data: [
            {
              id: "124",
              attributes: {
                name: "La Bruja.jpeg",
                mime: "image/jpeg",
                size: 124.35,
                width: 1280,
                height: 853,
                createdAt: "2023-01-26T10:34:26.058Z",
                url: "https://stomsdmediadev.blob.core.windows.net/msd-media-dev/assets-local-sdial/La%20Bruja.jpeg",
                ext: ".jpeg",
              },
            },
            {
              id: "122",
              attributes: {
                name: "Banana Mama Rooftop & Kitchen 5.jpeg",
                mime: "image/jpeg",
                size: 68.84,
                width: 960,
                height: 960,
                createdAt: "2023-01-26T10:30:22.701Z",
                url: "https://stomsdmediadev.blob.core.windows.net/msd-media-dev/assets-local-sdial/Banana%20Mama%20Rooftop%20%26%20Kitchen%205.jpeg",
                ext: ".jpeg",
              },
            },
            {
              id: "121",
              attributes: {
                name: "Sichuan Hot Pot.jpeg",
                mime: "image/jpeg",
                size: 2233.13,
                width: 6962,
                height: 4615,
                createdAt: "2023-01-18T17:39:30.518Z",
                url: "https://stomsdmediadev.blob.core.windows.net/msd-media-dev/assets-local-sdial/Sichuan%20Hot%20Pot.jpeg",
                ext: ".jpeg",
              },
            },
            {
              id: "120",
              attributes: {
                name: "mahbod-akhzami-kg40JrEsCvY-unsplash.jpg",
                mime: "image/jpeg",
                size: 1788.48,
                width: 6000,
                height: 4000,
                createdAt: "2023-01-18T17:39:29.900Z",
                url: "https://stomsdmediadev.blob.core.windows.net/msd-media-dev/assets-local-sdial/mahbod-akhzami-kg40JrEsCvY-unsplash.jpg",
                ext: ".jpg",
              },
            },
            {
              id: "119",
              attributes: {
                name: "Paella in Valencia.jpeg",
                mime: "image/jpeg",
                size: 1026.93,
                width: 3851,
                height: 2528,
                createdAt: "2023-01-18T17:39:29.720Z",
                url: "https://stomsdmediadev.blob.core.windows.net/msd-media-dev/assets-local-sdial/Paella%20in%20Valencia.jpeg",
                ext: ".jpeg",
              },
            },
            {
              id: "118",
              attributes: {
                name: "Rusconi's American Kitchen 3.jpeg",
                mime: "image/jpeg",
                size: 209.81,
                width: 1999,
                height: 1333,
                createdAt: "2023-01-18T17:39:29.502Z",
                url: "https://stomsdmediadev.blob.core.windows.net/msd-media-dev/assets-local-sdial/Rusconi%27s%20American%20Kitchen%203.jpeg",
                ext: ".jpeg",
              },
            },
            {
              id: "117",
              attributes: {
                name: "Quince 5.jpeg",
                mime: "image/jpeg",
                size: 197.32,
                width: 2000,
                height: 1500,
                createdAt: "2023-01-18T17:39:29.310Z",
                url: "https://stomsdmediadev.blob.core.windows.net/msd-media-dev/assets-local-sdial/Quince%205.jpeg",
                ext: ".jpeg",
              },
            },
            {
              id: "116",
              attributes: {
                name: "Rusconi's American Kitchen 2.jpeg",
                mime: "image/jpeg",
                size: 72.49,
                width: 1000,
                height: 1500,
                createdAt: "2023-01-18T17:39:29.221Z",
                url: "https://stomsdmediadev.blob.core.windows.net/msd-media-dev/assets-local-sdial/Rusconi%27s%20American%20Kitchen%202.jpeg",
                ext: ".jpeg",
              },
            },
            {
              id: "115",
              attributes: {
                name: "Pur' - Jean-Fran√ßois Rouquette 3.jpeg",
                mime: "image/jpeg",
                size: 63.2,
                width: 1280,
                height: 853,
                createdAt: "2023-01-18T17:39:29.039Z",
                url: "https://stomsdmediadev.blob.core.windows.net/msd-media-dev/assets-local-sdial/Pur%27%20-%20Jean-Fran%C3%A7ois%20Rouquette%203.jpeg",
                ext: ".jpeg",
              },
            },
            {
              id: "114",
              attributes: {
                name: "muneeb-muhamed-kPXT9HA4LOk-unsplash.jpg",
                mime: "image/jpeg",
                size: 516.64,
                width: 3456,
                height: 4608,
                createdAt: "2023-01-18T17:39:28.922Z",
                url: "https://stomsdmediadev.blob.core.windows.net/msd-media-dev/assets-local-sdial/muneeb-muhamed-kPXT9HA4LOk-unsplash.jpg",
                ext: ".jpg",
              },
            },
          ],
          meta: {
            pagination: {
              total: 122,
              pageSize: 10,
              pageCount: 13,
              page: 1,
            },
          },
        },
      },
    },
  },
];
