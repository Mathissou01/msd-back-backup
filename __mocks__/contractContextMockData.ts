import {
  Enum_Contract_Clienttype,
  Enum_Contract_Contractstatus,
} from "../src/graphql/codegen/generated-types";

export const mockData = {
  id: "1",
  attributes: {
    clientName: "Grand test",
    clientType: "city" as Enum_Contract_Clienttype,
    contractStatus: "Initialisation" as Enum_Contract_Contractstatus,
    siret: 12345678910123,
    clear: 123456789,
    ccap: 123456789,
    isNonExclusive: false,
    isRVFrance: true,
    pathId: 2,
    logo: {
      data: null,
    },
    channelType: {
      data: {
        id: "1",
        attributes: {
          hasWebApp: false,
          hasWebSite: false,
        },
      },
    },
    clientContact: {
      data: {
        id: "1",
        attributes: {
          firstName: "test",
          lastName: "test",
          email: "test.test@mail.com",
          phoneNumber: "07 81 21 56 81",
        },
      },
    },
    contractCustomization: {
      data: {
        id: "1",
        attributes: {
          primaryColor: "#9BCD41",
          secondaryColor: "#FFC229",
          textContrast: "#030F40",
        },
      },
    },
    editorialService: {
      data: {
        id: "1",
        attributes: {
          eventSubService: {
            data: {
              id: "1",
              attributes: {
                name: "Evénement ",
                isActivated: true,
              },
            },
          },
          freeContentSubServices: {
            data: [
              {
                id: "1",
                attributes: {
                  name: "Editorial 1",
                  isActivated: true,
                },
              },
              {
                id: "2",
                attributes: {
                  name: "Editorial 2",
                  isActivated: true,
                },
              },
            ],
          },
          newsSubService: {
            data: {
              id: "1",
              attributes: {
                name: "Actualités",
                isActivated: true,
              },
            },
          },
          quizSubService: {
            data: {
              id: "1",
              attributes: {
                name: "Quiz",
                isActivated: true,
              },
            },
          },
          tipSubService: {
            data: {
              id: "1",
              attributes: {
                name: "Astuces",
                isActivated: true,
              },
            },
          },
        },
      },
    },
    recyclingGuideService: {
      data: {
        id: "1",
        attributes: {
          name: "Guide du tri",
          isActivated: false,
          memoName: "Memo Tri",
        },
      },
    },
    pickUpDayService: {
      data: {
        id: "1",
        attributes: {
          name: "Jour de collecte",
          isActivated: false,
        },
      },
    },
    dropOffMapService: {
      data: {
        id: "1",
        attributes: {
          name: "Points d'apports et carte",
          isActivated: false,
        },
      },
    },
    requestService: {
      data: {
        id: "1",
        attributes: {
          name: "Demande",
          isActivated: false,
        },
      },
    },
  },
};
