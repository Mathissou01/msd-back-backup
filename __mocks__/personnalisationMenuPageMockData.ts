import { GetContractMenuByContractIdDocument } from "../src/graphql/codegen/generated-types";

export const defaultMockData = [
  {
    request: {
      query: GetContractMenuByContractIdDocument,
      variables: {
        contractId: "0",
      },
    },
    result: {
      data: {
        contract: {
          data: {
            attributes: {
              contractMenu: {
                data: {
                  id: "3",
                  attributes: {
                    serviceLinks: [
                      {
                        name: "Carte",
                        isDisplayed: true,
                        picto: {
                          data: null,
                        },
                      },
                      {
                        name: "Collecte à mon adresse !",
                        isDisplayed: true,
                        picto: {
                          data: null,
                        },
                      },
                      {
                        name: "Guide du tri",
                        isDisplayed: false,
                        picto: {
                          data: null,
                        },
                      },
                      {
                        name: "Faire une demande",
                        isDisplayed: true,
                        picto: {
                          data: null,
                        },
                      },
                      {
                        name: "Réduire mes déchets",
                        isDisplayed: false,
                        picto: {
                          data: null,
                        },
                      },
                      {
                        name: "Valoriser mes déchets",
                        isDisplayed: true,
                        picto: {
                          data: null,
                        },
                      },
                      {
                        name: null,
                        isDisplayed: false,
                        picto: {
                          data: null,
                        },
                        externalLink: null,
                      },
                      {
                        name: null,
                        isDisplayed: false,
                        picto: {
                          data: null,
                        },
                      },
                      {
                        name: null,
                        isDisplayed: false,
                        picto: {
                          data: null,
                        },
                      },
                      {
                        name: null,
                        isDisplayed: false,
                        picto: {
                          data: null,
                        },
                      },
                      {
                        name: null,
                        isDisplayed: false,
                        picto: {
                          data: null,
                        },
                      },
                      {
                        name: null,
                        isDisplayed: false,
                        picto: {
                          data: null,
                        },
                      },
                    ],
                  },
                },
              },
            },
          },
        },
      },
    },
  },
];
