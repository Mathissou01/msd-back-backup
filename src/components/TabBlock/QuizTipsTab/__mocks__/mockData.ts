import { GetQuizAndTipsBlockDocument } from "../../../../graphql/codegen/generated-types";

export const defaultMockData = [
  {
    request: {
      query: GetQuizAndTipsBlockDocument,
      variables: {
        contractId: "1",
      },
    },
    result: {
      data: {
        contractCustomizations: {
          data: [
            {
              attributes: {
                homepage: {
                  data: {
                    attributes: {
                      quizAndTipsBlock: {
                        data: {
                          id: "1",
                          attributes: {
                            title: "Titre modifi√©",
                            displayBlock: false,
                            displayQuiz: false,
                            quiz: {
                              data: {
                                id: "2",
                                attributes: {
                                  title: "Nom du quiz",
                                  publishedAt: "2022-11-20T02:34:14.535Z",
                                },
                              },
                            },
                            displayTips: false,
                            tips: {
                              data: [],
                            },
                          },
                        },
                      },
                    },
                  },
                },
              },
            },
          ],
        },
        services: {
          data: [
            {
              attributes: {
                serviceInstance: [
                  {
                    __typename: "ComponentMsdEditorial",
                    id: "4",
                    editorialServices: {
                      data: [
                        {
                          attributes: {
                            subServiceInstance: [
                              {
                                __typename: "ComponentEditoQuizzesSubService",
                                isActivated: true,
                                quizzes: {
                                  data: [
                                    {
                                      id: "2",
                                      attributes: {
                                        title: "Nom du quiz",
                                        publishedAt: "2022-11-20T02:34:14.535Z",
                                      },
                                    },
                                    {
                                      id: "1",
                                      attributes: {
                                        title: "truc",
                                        publishedAt: "2022-11-18T10:16:08.936Z",
                                      },
                                    },
                                  ],
                                },
                              },
                            ],
                          },
                        },
                        {
                          attributes: {
                            subServiceInstance: [
                              {
                                __typename: "ComponentEditoTipsSubService",
                                isActivated: true,
                                tips: {
                                  data: [
                                    {
                                      id: "5",
                                      attributes: {
                                        title: "nah",
                                        publishedAt: "2022-11-19T00:14:37.403Z",
                                      },
                                    },
                                    {
                                      id: "2",
                                      attributes: {
                                        title: "Titre de l'astuce 1",
                                        publishedAt: "2022-11-19T00:14:08.232Z",
                                      },
                                    },
                                    {
                                      id: "1",
                                      attributes: {
                                        title: "Titre de l'astuce 2",
                                        publishedAt: "2022-11-18T10:17:11.158Z",
                                      },
                                    },
                                    {
                                      id: "3",
                                      attributes: {
                                        title: "Titre de l'astuce 3",
                                        publishedAt: "2022-11-19T00:14:20.490Z",
                                      },
                                    },
                                    {
                                      id: "4",
                                      attributes: {
                                        title: "Titre de l'astuce 4",
                                        publishedAt: "2022-11-19T00:14:30.273Z",
                                      },
                                    },
                                  ],
                                },
                              },
                            ],
                          },
                        },
                      ],
                    },
                  },
                ],
              },
            },
          ],
        },
      },
    },
  },
];

export const errorMockData = [
  {
    request: {
      query: GetQuizAndTipsBlockDocument,
      variables: {
        contractId: "1",
      },
    },
    error: new Error("An error occured"),
  },
];
