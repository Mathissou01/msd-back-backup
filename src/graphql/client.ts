import {
  ApolloClient,
  InMemoryCache,
  NormalizedCacheObject,
} from "@apollo/client";

let client: ApolloClient<NormalizedCacheObject>;

if (process.env.NEXT_PUBLIC_MOCK === "true") {
  /* With .env variable NEXT_PUBLIC_MOCK=true, return mock graphql client with fake response data */
  client = (await import("./mockedClient")).default(1000);
} else {
  client = await new ApolloClient({
    uri: `${process.env.NEXT_PUBLIC_API_URL}/graphql`,
    cache: new InMemoryCache({
      typePolicies: {
        UploadFolder: {
          merge: true,
        },
        UploadFile: {
          merge: true,
        },
        Contract: {
          merge: true,
        },
        ContractCustomization: {
          merge: true,
        },
        New: {
          merge: true,
        },
        Tip: {
          merge: true,
        },
        RecyclingGuideService: {
          merge: true,
        },
        WasteForm: {
          merge: true,
        },
        WasteFamily: {
          merge: true,
        },
        Flow: {
          merge: true,
        },
        FlowColor: {
          merge: true,
        },
        DropOffMap: {
          merge: true,
        },
        Request: {
          merge: true,
        },
        RequestAggregate: {
          merge: true,
        },
      },
    }),
  });
}

export default client;
