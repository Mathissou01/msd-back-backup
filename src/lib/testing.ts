import { prettyDOM } from "@testing-library/react";

export function removeUnstableHtmlPropertiesForTesting(
  htmlElement: HTMLElement,
  attributeName: string,
) {
  const domHTML = prettyDOM(htmlElement, Infinity).toString();
  if (!domHTML) return undefined;
  const regex = new RegExp(`${attributeName}(.*)"(.*)"`, "g");
  return domHTML.replaceAll(regex, "");
}
