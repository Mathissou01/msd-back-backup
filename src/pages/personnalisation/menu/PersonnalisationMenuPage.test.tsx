import { render, screen } from "@testing-library/react";
import { MockedProvider } from "@apollo/client/testing";
import React from "react";
import { defaultMockData } from "../../../../__mocks__/personnalisationMenuPageMockData";
import PersonnalisationMenuPage from "./index.page";

describe("PersonnalisationMenuPage", () => {
  it("renders loading and loaded state", async () => {
    const { container } = render(
      <MockedProvider mocks={defaultMockData} addTypename={false}>
        <PersonnalisationMenuPage />
      </MockedProvider>,
    );

    expect(await screen.findByTestId("common-spinner")).toBeInTheDocument();
    expect(await container).toMatchSnapshot();
    expect(
      await screen.findByText("Ordre d’affichage du menu"),
    ).toBeInTheDocument();
    // expect(
    //   await screen.findByText("Collecte à mon adresse !"),
    // ).toBeInTheDocument();
    expect(await container).toMatchSnapshot();
  });
});
