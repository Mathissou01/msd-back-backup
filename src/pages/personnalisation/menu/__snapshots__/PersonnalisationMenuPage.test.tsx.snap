// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PersonnalisationMenuPage renders loading and loaded state 1`] = `
<div>
  <hgroup
    class="c-PageTitle"
  >
    <h1
      data-testid="title"
    >
      Menu
    </h1>
    <p
      class="c-PageTitle__Description"
      data-testid="description"
    >
      Choisissez, personnalisez et ordonnez les entrées du menu de gauche
    </p>
  </hgroup>
  <div
    class="c-PersonnalisationMenuPage"
  >
    <div
      class="c-CommonLoader c-CommonLoader_cover c-CommonLoader_grow"
    >
      <div
        class="c-CommonSpinner"
        data-testid="common-spinner"
      >
        <div />
        <div />
        <div />
        <div />
      </div>
    </div>
  </div>
</div>
`;

exports[`PersonnalisationMenuPage renders loading and loaded state 2`] = `
<div>
  <hgroup
    class="c-PageTitle"
  >
    <h1
      data-testid="title"
    >
      Menu
    </h1>
    <p
      class="c-PageTitle__Description"
      data-testid="description"
    >
      Choisissez, personnalisez et ordonnez les entrées du menu de gauche
    </p>
  </hgroup>
  <div
    class="c-PersonnalisationMenuPage"
  >
    <form
      class="c-PersonnalisationMenuPage__Form"
    >
      <div
        class="c-PersonnalisationMenuPage__Group"
      >
        <h2
          class="c-PersonnalisationFooterPage__Title"
        >
          Menu
        </h2>
        <div
          class="c-FormServiceLinks"
        >
          <div
            class="c-FormLabel"
          >
            <label
              class="c-FormLabel__LabelWrapper"
            >
              <span
                class="c-FormLabel__Label"
              >
                Ordre d’affichage du menu
              </span>
            </label>
          </div>
        </div>
      </div>
      <div
        class="c-PersonnalisationMenuPage__Buttons"
      >
        <button
          class="c-CommonButton c-CommonButton_disabled c-CommonButton_primary c-CommonButton_fontSmall"
          disabled=""
          type="submit"
        >
          <div
            class="c-CommonButton__Picto c-CommonButton__Picto_check"
          />
          Enregistrer les modifications
        </button>
        <button
          class="c-CommonButton c-CommonButton_disabled c-CommonButton_secondary c-CommonButton_fontSmall"
          disabled=""
          type="button"
        >
          <div
            class="c-CommonButton__Picto c-CommonButton__Picto_cross"
          />
          Annuler les modifications
        </button>
      </div>
    </form>
  </div>
</div>
`;
